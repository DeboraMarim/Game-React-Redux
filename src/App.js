import React from 'react';
import { Route, Switch } from 'react-router-dom';
import Login from './Pages/Login';
import Game from './Pages/Game';
import Config from './Pages/Config';
import Feedback from './Pages/Feedback';
import Ranking from './Pages/Ranking';

export default function App() {
  return (
    <div>
      <Switch>
        <Route exact path="/" component={ Login } />
        <Route exact path="/game" component={ Game } />
        <Route exact path="/config" component={ Config } />
        <Route exact path="/feedback" component={ Feedback } />
        <Route exact path="/ranking" component={ Ranking } />
      </Switch>
    </div>
  );
}
