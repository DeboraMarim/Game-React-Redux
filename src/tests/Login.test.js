import React from 'react';
import userEvent from '@testing-library/user-event';
import { screen, waitFor } from '@testing-library/react';
import App from '../App';
import { renderWithRouterAndRedux } from './helpers/renderWithRouterAndRedux';

// jest.mock('../Redux/Actions', () => ({
//   addEmail: jest.fn(),
//   addName: jest.fn(),
// }));
const gravatar = 'input-gravatar-email';
describe('Login', () => {
  it('deve renderizar o componente corretamente', () => {
    const { getByTestId } = renderWithRouterAndRedux(<App />);
    expect(getByTestId('input-player-name')).toBeInTheDocument();
    expect(getByTestId(gravatar)).toBeInTheDocument();
    expect(getByTestId('btn-play')).toBeInTheDocument();
    expect(getByTestId('btn-settings')).toBeInTheDocument();
  });
  it('verificação de renderização da page Game após click', async () => {
    const { history } = renderWithRouterAndRedux(<App />);
    const nameInput = screen.getByRole('textbox', { name: /name/i });
    const emailInput = screen.getByTestId(gravatar);

    userEvent.type(nameInput, 'User');
    userEvent.type(emailInput, 'user@email.com');
    const playButton = screen.getByRole('button', { name: /play/i });

    userEvent.click(playButton);

    await waitFor(() => {
      expect(history.location.pathname).toBe('/game');
    });
  });
  it('verificação de renderização da page Config após click', async () => {
    const { history } = renderWithRouterAndRedux(<App />);
    const configButton = screen.getByRole('button', { name: /configurações/i });
    expect(configButton).toBeInTheDocument();
    userEvent.click(configButton);
    await waitFor(() => {
      expect(history.location.pathname).toBe('/config');
    });
  });
});
