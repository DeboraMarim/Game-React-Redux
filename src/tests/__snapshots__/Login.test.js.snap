// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<Memo(Connect(Login))
  dispatch={[MockFunction]}
  history={
    Object {
      "push": [MockFunction],
    }
  }
/>
`;
